plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -351,7 +353,18 @@ else if (precision > Decimals.MAX_PRECISION || actualPrecision <= 0) {
 
             // This mapping covers both DATE and TIMESTAMP, as Oracle's DATE has second precision.
             case OracleTypes.TIMESTAMP:
-                return Optional.of(oracleTimestampColumnMapping());
+                if ("DATE".equalsIgnoreCase(typeHandle.getJdbcTypeName().orElseThrow(() -> new IllegalStateException("type name not present")))) {
+                    return Optional.of(ColumnMapping.longMapping(
+                            TIMESTAMP_SECONDS,
+                            OracleClient::oracleTimestampReadFunction,
+                            trinoTimestampToOracleDateWriteFunction(),
+                            FULL_PUSHDOWN));
+                }
+                return Optional.of(ColumnMapping.longMapping(
+                        TIMESTAMP_MILLIS,
+                        OracleClient::oracleTimestampReadFunction,
+                        trinoTimestampToOracleTimestampWriteFunction(),
```

Comment:
read function is a method handle, write function has a factory method returning the actual write function
is this inconsistency intentional?